src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -92,7 +92,14 @@ public static bool IsMsBuildBasedProject(string projectFullPath)
 
                 using (var process = Process.Start(processStartInfo))
                 {
-                    process.WaitForExit(MsBuildWaitTime);
+                    var finished = process.WaitForExit(MsBuildWaitTime);
+
+                    if (!finished)
+                    {
+                        process.Kill();
```

Comment:
This should probably go in a try/catch. There is a potential race case here in that WaitForExit could return false because the process is still going, but by the time process.Kill is reached, the process has exited, which will then throw an invalid operation exception. That error would just be confusing.